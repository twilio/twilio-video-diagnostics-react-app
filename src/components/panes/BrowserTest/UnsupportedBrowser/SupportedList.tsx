import { makeStyles, Typography, Table, TableHead, TableBody, TableRow, TableCell, Link } from '@material-ui/core';

const useStyles = makeStyles({
  tableHead: {
    background: '#F4F4F6',
  },
  table: {
    '& td': {
      padding: '0.9em',
    },
  },
});

export function SupportedList() {
  const classes = useStyles();

  return (
    <>
      <Table className={classes.table}>
        <TableHead className={classes.tableHead}>
          <TableRow>
            <TableCell>
              <Typography variant="body1">
                <strong>Supported browsers</strong>
              </Typography>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          <TableRow>
            <TableCell>
              <Typography variant="body1">
                <Link
                  color="inherit"
                  underline="always"
                  target="_blank"
                  rel="noopener"
                  href="https://www.google.com/chrome/"
                >
                  Chrome
                </Link>
              </Typography>
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell>
              <Typography variant="body1">
                <Link
                  color="inherit"
                  underline="always"
                  target="_blank"
                  rel="noopener"
                  href="https://www.mozilla.org/en-US/firefox/new/"
                >
                  Firefox
                </Link>
              </Typography>
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell>
              <Typography variant="body1">
                <Link
                  color="inherit"
                  underline="always"
                  target="_blank"
                  rel="noopener"
                  href="https://support.apple.com/downloads/safari"
                >
                  Safari
                </Link>
              </Typography>
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell style={{ borderBottom: 'none' }}>
              <Typography variant="body1">
                <Link
                  color="inherit"
                  underline="always"
                  target="_blank"
                  rel="noopener"
                  href="https://www.microsoft.com/en-us/edge"
                >
                  Edge
                </Link>
              </Typography>
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </>
  );
}
