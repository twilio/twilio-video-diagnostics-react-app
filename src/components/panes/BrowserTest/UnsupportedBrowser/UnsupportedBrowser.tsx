import {
  makeStyles,
  TextField,
  Container,
  Grid,
  Typography,
  Paper,
  InputAdornment,
  IconButton,
  Link,
} from '@material-ui/core';
import { CopyIcon } from '../../../../icons/CopyIcon';
import { ErrorIcon } from '../../../../icons/ErrorIcon';
import { SupportedList } from './SupportedList';

const useStyles = makeStyles({
  paper: {
    padding: '1.5em',
    borderRadius: '8px',
    width: '388px',
  },
  paperContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  copyButton: {
    borderLeft: 'solid 1px #E1E3EA',
    maxHeight: '2.5em',
    display: 'flex',
  },
  errorIcon: {
    position: 'absolute',
    right: 'calc(100% + 18px)',
  },
  caption: {
    marginTop: '1.3em',
  },
});

export function UnsupportedBrowser() {
  const classes = useStyles();

  const appURL = window.location.href;

  const copyAppLink = () => {
    navigator.clipboard.writeText(appURL);
  };

  return (
    <>
      <Container>
        <Grid container alignItems="center" justifyContent="space-between">
          <Grid item md={5}>
            <Typography variant="h1" gutterBottom style={{ position: 'relative' }}>
              <div className={classes.errorIcon}>
                <ErrorIcon />
              </div>{' '}
              New browser needed
            </Typography>
            <Typography variant="body1" gutterBottom>
              Oh no, your browser isn't supported! Download and install a new one from the list, copy this url and paste
              it into your new browser to restart the test.
            </Typography>
            <TextField
              fullWidth={true}
              id="read-only-app-link"
              defaultValue={appURL}
              variant="outlined"
              size="small"
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <div className={classes.copyButton}>
                      <IconButton edge="end" onClick={copyAppLink}>
                        <CopyIcon />
                      </IconButton>
                    </div>
                  </InputAdornment>
                ),
                readOnly: true,
              }}
            />
          </Grid>
          â€‹
          <Grid item md={5} className={classes.paperContainer}>
            <Paper className={classes.paper}>
              <SupportedList />
            </Paper>
            <Typography variant="caption" align="center" className={classes.caption}>
              We support the most recent and the two previous versions. Please{' '}
              <Link
                color="inherit"
                underline="always"
                target="_blank"
                rel="noopener"
                href="https://www.twilio.com/docs/video/javascript#supported-browsers"
              >
                visit our docs
              </Link>{' '}
              for more details.
            </Typography>
          </Grid>
        </Grid>
      </Container>
    </>
  );
}
