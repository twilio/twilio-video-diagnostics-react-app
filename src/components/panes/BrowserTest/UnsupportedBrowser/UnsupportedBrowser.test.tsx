import { mount, shallow } from 'enzyme';
import { IconButton } from '@material-ui/core';
import { UnsupportedBrowser } from './UnsupportedBrowser';

const mockAppURL = 'https://mockurl.com';

// @ts-ignore
delete window.location;

// @ts-ignore
window.location = {
  href: mockAppURL,
};

const mockWriteText = jest.fn();

Object.defineProperty(navigator, 'clipboard', {
  value: {
    writeText: mockWriteText,
  },
});

describe('the UnsupportedBrowser component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UnsupportedBrowser />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should copy the app url when copy button is clicked on', () => {
    const wrapper = mount(<UnsupportedBrowser />);
    const copyButton = wrapper.find(IconButton);

    copyButton.simulate('click');

    expect(navigator.clipboard.writeText).toHaveBeenCalledWith(mockAppURL);
  });
});
