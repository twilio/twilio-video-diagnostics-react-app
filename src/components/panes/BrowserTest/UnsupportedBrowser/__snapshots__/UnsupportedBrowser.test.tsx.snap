// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UnsupportedBrowser component should render correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          style={
            Object {
              "position": "relative",
            }
          }
          variant="h1"
        >
          <div
            className="makeStyles-errorIcon-4"
          >
            <ErrorIcon />
          </div>
           
          New browser needed
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="body1"
        >
          Oh no, your browser isn't supported! Download and install a new one from the list, copy this url and paste it into your new browser to restart the test.
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(TextField))
          InputProps={
            Object {
              "endAdornment": <WithStyles(ForwardRef(InputAdornment))
                position="end"
              >
                <div
                  className="makeStyles-copyButton-3"
                >
                  <WithStyles(ForwardRef(IconButton))
                    edge="end"
                    onClick={[Function]}
                  >
                    <CopyIcon />
                  </WithStyles(ForwardRef(IconButton))>
                </div>
              </WithStyles(ForwardRef(InputAdornment))>,
              "readOnly": true,
            }
          }
          defaultValue="https://mockurl.com"
          fullWidth={true}
          id="read-only-app-link"
          size="small"
          variant="outlined"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-paperContainer-2"
        item={true}
        md={5}
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-paper-1"
        >
          <SupportedList />
        </WithStyles(ForwardRef(Paper))>
        <WithStyles(ForwardRef(Typography))
          align="center"
          className="makeStyles-caption-5"
          variant="caption"
        >
          We support the most recent and the two previous versions. Please
           
          <WithStyles(ForwardRef(Link))
            color="inherit"
            href="https://www.twilio.com/docs/video/javascript#supported-browsers"
            rel="noopener"
            target="_blank"
            underline="always"
          >
            visit our docs
          </WithStyles(ForwardRef(Link))>
           
          for more details.
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
